src/Naderman/Composer/AWS/AwsClient.php

```diff
@@ -35,6 +36,16 @@ class AwsClient
     protected $io;
 
     /**
+     * @var S3Client
+     */
+    protected $client;
```

Comment:
why is this needed as state now? The s3factory function still returns the client, so I'd rather it wasn't state of this object?